pipeline/stage03_trigger_detection/scripts/minima.py

```diff
@@ -6,17 +6,41 @@
 from utils import load_neo, write_neo, remove_annotations
 
 
-def detect_minima(asig, order):
+def detect_minima(asig, order, interpolation_points, sampling_time, interpolation):
     signal = asig.as_array()
+    
     t_idx, channel_idx = argrelmin(signal, order=order, axis=0)
 
-    sort_idx = np.argsort(t_idx)
+    if int(interpolation) == 0:
+        fitted_idx = []
+        fitted_idx_times = []
 
-    evt = neo.Event(times=asig.times[t_idx[sort_idx]],
-                     labels=['UP'] * len(t_idx),
-                     name='Transitions',
-                     minima_order=order,
-                     array_annotations={'channels':channel_idx[sort_idx]})
+        for i in range(0,len(t_idx)):
+        
+            if t_idx[i] - int((interpolation_points-1)/2) >= 0 and t_idx[i] + int((interpolation_points-1)/2)+1 <= len(signal[:,0]):
```

Comment:
Try to simplify your code. That makes it also easier to spot errors.
For example:
```
start = t_i - int(interpolation_points/2)
stop = start + int(interpolation_points)

if start < 0 or stop > len(signal):
    minimum = t_i
```